lib/utils.pm

```diff
@@ -1022,4 +1023,22 @@ sub svirt_host_basedir {
     return get_var('VIRSH_OPENQA_BASEDIR', '/var/lib');
 }
 
+sub prepare_ssh_localhost_key_login {
+    # in case localhost is already inside known_hosts
+    if (script_run('test -e ~/.ssh') == 0) {
+        assert_script_run('ls -l ~/.ssh');
+        assert_script_run('ssh-keygen -R localhost');
+    }
+
+    # generate ssh key for root
+    if (script_run('! test -e ~/.ssh/id_rsa') == 0) {
+        assert_script_run('ssh-keygen -t rsa -N "" -f ~/.ssh/id_rsa');
+    }
+
+    # add key to authorized_keys
+    if (script_run('! grep "$(cat ~/.ssh/id_rsa.pub)" ~/.ssh/authorized_keys') == 0) {
+        assert_script_run('cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys');
```

Comment:
why not just `assert_script_run 'key=$(cat ~/.ssh/id_rsa.pub); grep -q $key ~/.ssh/authorized_keys || echo $key >> ~/.ssh/authorized_keys`